### PR TITLE
drivers: clock control: stm32u5 wait for fixed clock to get freq

### PR DESCRIPTION
**Pull request by FRASTM**
Wait for the fixed clock ready flag before getting the sysclock freq

When running the _tests/kernel/timer/timer_api_  in the tickless (DEVICE_PM) configuration on the b_u585i_iot02a platform, it might happen that the start up frequency is not ready, after low power mode phase when trying to get it: MSIS or HSI
```
*** Booting Zephyr OS build zephyr-v3.3.0-686-g1be74ac6d71a ***
Running TESTSUITE timer_api
===================================================================
START - test_sleep_abs
PASS - test_sleep_abs in 0.013 seconds
===================================================================
START - test_time_conversions
PASS - test_time_conversions in 0.060 seconds
===================================================================
START - test_timeout_abs
ASSERTION FAIL [0] @ WEST_TOPDIR/zephyr/drivers/clock_control/clock_stm32_ll_u5.c:83
Unexpected startup freq
E: ***** HARD FAULT *****
E:   Fault escalation (see below)
E: ARCH_EXCEPT with reason 4
E: r0/a1:  0x00000004  r1/a2:  0x00000053  r2/a3:  0x00000001
E: r3/a4:  0x080041d9 r12/ip:  0x00000000 r14/lr:  0x08003f01
E:  xpsr:  0x41000053
E: Faulting instruction address (r15/pc): 0x08007540
E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
E: Fault during interrupt handling
E: Current thread: 0x200009e8 (idle)
E: Halting system
```

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

File: drivers/clock_control/clock_stm32_ll_u5.c [hunk 1]
```diff
@@ -74,6 +74,19 @@ static uint32_t get_pllsrc_frequency(size_t pll_id)
...
+		while (LL_RCC_HSI_IsReady() != 1) {
+		/* Wait for HSI ready */
+		}
+	}
+
+	if (IS_ENABLED(STM32_MSIS_ENABLED)) {
+		while (LL_RCC_MSI_IsReady() != 1) {
+		/* Wait for MSI ready */
+		}
+	}
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_u5.c:88]: I understand the problem in resume from low power, but will this work in usual boot ?
This function is called before set_up_fixed_clock_sources which enables these clocks

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/clock_control/clock_stm32_ll_u5.c (hunk 1 above)
Comment (FRASTM) [reply drivers/clock_control/clock_stm32_ll_u5.c:88]: It boots. However this part could still be flagged by `CONFIG_PM` to reduce impact 
Even when exiting the low power mode or booting the same function stm32_clock_control_init is called.

File: drivers/clock_control/clock_stm32_ll_u5.c [hunk 2]
```diff
@@ -79,6 +79,13 @@ static uint32_t get_startup_frequency(void)
 		return get_msis_frequency();
 	case LL_RCC_SYS_CLKSOURCE_STATUS_HSI:
 		return STM32_HSI_FREQ;
+	case LL_RCC_SYS_CLKSOURCE_STATUS_HSE:
+		/* HSE is un-expected freq */
+		return STM32_HSE_FREQ;
```
Comment (gautierg-st) [inline drivers/clock_control/clock_stm32_ll_u5.c:84]: Question: if HSE is unexpected, shouldn't we also have an __ASSERT()?

Comment (FRASTM) [reply drivers/clock_control/clock_stm32_ll_u5.c:84]: indeed, _ASSERT_ is the cause of crash "ASSERTION FAIL"

File: drivers/clock_control/clock_stm32_ll_u5.c [hunk 3]
```diff
@@ -79,6 +79,13 @@ static uint32_t get_startup_frequency(void)
 		return get_msis_frequency();
 	case LL_RCC_SYS_CLKSOURCE_STATUS_HSI:
 		return STM32_HSI_FREQ;
+	case LL_RCC_SYS_CLKSOURCE_STATUS_HSE:
+		/* HSE is un-expected freq */
```
Comment (erwango) [inline drivers/clock_control/clock_stm32_ll_u5.c:83]: Comment to be removed as this is now a tolerated case.

**Review by erwango — COMMENTED** [review]
Otherwise LGTM

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request closed without merging**